### PR TITLE
[onnx] Lowerings from `onnx.tan`

### PR DESCRIPTION
**Pull request by rsuderman**
Started work on the `tan` lowerings for ONNX to Torch. Uses `sin` and `cos` to represent a `tan`.

### CODE REVIEW

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp [hunk 1]
```diff
@@ -26,4 +26,20 @@ using namespace mlir::torch::onnx_c;
...
+      "Tan", 7, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value operand;
+        if (binder.tensorOperand(operand) ||
+            binder.tensorResultType(resultType))
+          return failure();
+        auto loc = binder.op->getLoc();
+        Value sin = rewriter.create<Torch::AtenSinOp>(loc, resultType, operand);
+        Value cos = rewriter.create<Torch::AtenCosOp>(loc, resultType, operand);
+        rewriter.replaceOpWithNewOp<Torch::AtenDivTensorOp>(
```
Comment (vivekkhandelwal1) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp:41]: You can directly create `AtenTanOp`.

**Review by vivekkhandelwal1 — CHANGES_REQUESTED** [review]

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp (hunk 1 above)
Comment (rsuderman) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp]: `AtenTanOp` had not been added and I was not certain whether it was an `aten` operation in `torch`. Found a reference and added `AtenTanOp` with the corresponding lowering to `math.tan`

File: lib/Conversion/TorchToLinalg/Uncategorized.cpp [hunk 2]
```diff
@@ -1311,7 +1315,7 @@ class ConvertElementwiseOp : public ConversionPattern {
   LogicalResult
   matchAndRewrite(Operation *op, ArrayRef<Value> operands,
                   ConversionPatternRewriter &rewriter) const override {
-    if (!isa<AtenTanhOp, AtenReluOp, AtenPreluOp, AtenGeluOp,
+    if (!isa<AtenTanOp, AtenTanhOp, AtenReluOp, AtenPreluOp, AtenGeluOp,
```
Comment (ramiro050) [inline lib/Conversion/TorchToLinalg/Uncategorized.cpp:1318]: You also need to add the op to

https://github.com/llvm/torch-mlir/blob/6fbd8ea0c8a97e57d03cd3598cef5941d14f93c4/lib/Conversion/TorchToLinalg/Uncategorized.cpp#L1966

**Review by ramiro050 — CHANGES_REQUESTED** [review]
Changes that add e2e support for new torch ops need dtype/shape functions and e2e tests. Similar to: https://github.com/llvm/torch-mlir/pull/2635

More info in the docs:
- https://github.com/llvm/torch-mlir/blob/main/docs/adding_abstract_interpretation_functions.md
- https://github.com/llvm/torch-mlir/blob/main/docs/adding_an_e2e_test.md

File: lib/Conversion/TorchToLinalg/Uncategorized.cpp (hunk 2 above)
Comment (rsuderman) [reply lib/Conversion/TorchToLinalg/Uncategorized.cpp:1318]: I believe I have the remaining boilerplate based on the sample PR.

**Review by ramiro050 — APPROVED** [review]
Thanks!

**Pull request merged**